pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -476,44 +473,77 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 			return nil
 		}
 		peerGatewayIP := ip.NextIP(peerPodCIDRAddr)
-		peerConfig[peerPodCIDR] = peerGatewayIP
-		podCIDRs = append(podCIDRs, peerPodCIDR)
+		peerConfigs[peerPodCIDR] = peerGatewayIP
+		peerPodCIDRs = append(peerPodCIDRs, peerPodCIDR)
+		peerNodeIP := peerNodeIPs.IPv4
+		family := "IPv4"
+		if peerGatewayIP.To4() == nil {
+			peerNodeIP = peerNodeIPs.IPv6
+			family = "IPv6"
+		}
+
+		klog.InfoS("Adding route and flow to Node", "Node", nodeName, "podCIDR", podCIDR,
+			"peerNodeIP", peerNodeIP, "IPFamily", family)
 	}
 
-	ipsecTunOFPort := int32(0)
+	var ipsecTunOFPorts []uint32
 	if c.networkConfig.EnableIPSecTunnel {
 		// Create a separate tunnel port for the Node, as OVS IPSec monitor needs to
 		// read PSK and remote IP from the Node's tunnel interface to create IPSec
 		// security policies.
-		if ipsecTunOFPort, err = c.createIPSecTunnelPort(nodeName, peerNodeIP); err != nil {
+		createIPSecTunnelPort := func(ip net.IP) (uint32, error) {
+			if ip == nil {
+				return 0, nil
+			}
+			ipsecTunOFPort := int32(0)
+			if ipsecTunOFPort, err = c.createIPSecTunnelPort(nodeName, ip); err != nil {
```

Comment:
I checked `createIPSecTunnelPort`, it calculates the tunnel port name based on `nodeName` and will think the IP has changed if calling it twice. So this is not going to work. I'm not sure if we really need to two ports in a dual-stack cluster. @jianjuns do you know whether it can work if we just create IPv4 tunnel in dual-stack cluster?

Reply:
Good catch! I think we just need one tunnel port, which should use a single destination IP either v4 or v6.

Reply:
done